src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -72,6 +72,10 @@ function SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver
     _boundary_conditions = digest_boundary_conditions(boundary_conditions, mesh, solver,
                                                       cache)
 
+    if mesh isa TreeMesh || mesh isa StructuredMesh
+        check_periodicity_mesh_boundary_conditions(mesh, _boundary_conditions)
+    end
```

Comment:
I don't really like constructions like these since they are not extensible from the outside. Can't we just call

```suggestion
    check_periodicity_mesh_boundary_conditions(mesh, _boundary_conditions)
```

here and implement a dummy (general) method for the other mesh types?